### PR TITLE
voiceassistant: only commit the spoken words in the chat context

### PR DESCRIPTION
**Pull request by theomonnom**
This is useful when the speech was interrupted. This makes sure the LLM has the same "context" as what the user heard

### CODE REVIEW

File: livekit-agents/livekit/agents/voice_assistant/voice_assistant.py [hunk 1]
```diff
@@ -637,12 +637,15 @@ def _commit_user_question_if_needed() -> None:
...
 
-                collected_text = answer_synthesis.collected_text
+                collected_text = answer_synthesis.tr_fwd.played_text
                 interrupted = answer_synthesis.interrupted
 
         if speech_info.add_to_chat_ctx and (not user_question or user_speech_committed):
             self._chat_ctx.messages.extend(extra_tools_messages)
 
+            if interrupted:
+                collected_text += "..."
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:647]: This is kind of opinionated, wdyt?

Comment (nbsp) [reply livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:647]: maybe the singular ellipsis `…` instead of `...` works better?

Comment (davidzhao) [reply livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:647]: I think it's fine.. perhaps make this a constant? (if folks want to customize, we could offer it as an option)

File: livekit-agents/livekit/agents/voice_assistant/agent_output.py [hunk 2]
```diff
@@ -34,13 +34,16 @@ def __init__(
...
         self._interrupt_fut = asyncio.Future[None]()
-        self._collected_text = ""  # collected text from the async stream
         self._speech_id = speech_id
 
     @property
     def speech_id(self) -> str:
         return self._speech_id
 
+    @property
+    def tr_fwd(self) -> agent_transcription.TTSSegmentsForwarder:
```
Comment (nbsp) [inline livekit-agents/livekit/agents/voice_assistant/agent_output.py:44]: make this more understandable in the public API at least

```suggestion
    def transcription_forwarder(self) -> agent_transcription.TTSSegmentsForwarder:
```

**Review by davidzhao — APPROVED** [review]

File: livekit-agents/livekit/agents/voice_assistant/agent_output.py (hunk 2 above)
Comment (davidzhao) [reply livekit-agents/livekit/agents/voice_assistant/agent_output.py:44]: good point, +1 making the public API clear vs abbreviated

Comment (theomonnom) [reply livekit-agents/livekit/agents/voice_assistant/agent_output.py:44]: this isn't public, this is still internal

Comment (davidzhao) [reply livekit-agents/livekit/agents/voice_assistant/agent_output.py:44]: maybe `tts_forwarder` ?

File: livekit-agents/livekit/agents/voice_assistant/voice_assistant.py (hunk 1 above)
Comment (theomonnom) [reply livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:647]: if folks want to customize it, they can listen to the agent_speech_interrupted event

Comment (Toubat) [reply livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:657]: I'd like this to be customized :)

**Pull request merged**